extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/ItagItem.java

```diff
@@ -165,76 +201,311 @@ public MediaFormat getMediaFormat() {
+                ? audioChannels
+                : AUDIO_CHANNELS_NOT_APPLICABLE_OR_UNKNOWN;
+    }
+
+    /**
+     * Get the {@code targetDurationSec} value.
+     *
+     * <p>
+     * This value is the average time in seconds of the duration of sequences of livestreams and
+     * ended livestreams. It is only returned by YouTube for these stream types, and makes no sense
+     * for videos, so {@link #TARGET_DURATION_SEC_UNKNOWN} is returned for those.
+     * </p>
+     *
+     * @return the {@code targetDurationSec} value or {@link #TARGET_DURATION_SEC_UNKNOWN}
+     */
+    public int getTargetDurationSec() {
+        return targetDurationSec;
+    }
+
+    /**
+     * Set the {@code targetDurationSec} value.
+     *
+     * <p>
+     * This value is the average time in seconds of the duration of sequences of livestreams and
+     * ended livestreams.
+     * </p>
+     *
+     * <p>
+     * It is only returned for these stream types by YouTube and makes no sense for videos, so
+     * {@link #TARGET_DURATION_SEC_UNKNOWN} will be set/used for video streams or if this value is
+     * less than or equal to 0.
+     * </p>
+     *
+     * @param targetDurationSec the target duration of a segment of streams which are using the
+     *                          live delivery method type
+     */
+    public void setTargetDurationSec(final int targetDurationSec) {
+        this.targetDurationSec = targetDurationSec > 0
+                ? targetDurationSec
+                : TARGET_DURATION_SEC_UNKNOWN;
+    }
+
+    /**
+     * Get the {@code approxDurationMs} value.
+     *
+     * <p>
+     * It is only known for DASH progressive streams, so {@link #APPROX_DURATION_MS_UNKNOWN} is
+     * returned for other stream types or if this value is less than or equal to 0.
+     * </p>
+     *
+     * @return the {@code approxDurationMs} value or {@link #APPROX_DURATION_MS_UNKNOWN}
+     */
+    public long getApproxDurationMs() {
+        return approxDurationMs;
+    }
+
+    /**
+     * Set the {@code approxDurationMs} value.
+     *
+     * <p>
+     * It is only known for DASH progressive streams, so {@link #APPROX_DURATION_MS_UNKNOWN} is
+     * set/used for other stream types or if this value is less than or equal to 0.
+     * </p>
+     *
+     * @param approxDurationMs the approximate duration of a DASH progressive stream, in
+     *                         milliseconds
+     */
+    public void setApproxDurationMs(final long approxDurationMs) {
+        this.approxDurationMs = approxDurationMs > 0
+                ? approxDurationMs
+                : APPROX_DURATION_MS_UNKNOWN;
+    }
+
+    /**
+     * Get the {@code contentLength} value.
+     *
+     * <p>
+     * It is only known for DASH progressive streams, so {@link #CONTENT_LENGTH_UNKNOWN} is
+     * returned for other stream types or if this value is less than or equal to 0.
+     * </p>
+     *
+     * @return the {@code contentLength} value or {@link #CONTENT_LENGTH_UNKNOWN}
```

Comment:
Do not make changes, but, just saying, this javadoc could be **3 lines** long instead of 8
```suggestion
     * Get the {@code contentLength} value. It is only known for DASH progressive streams, so
     * {@link #CONTENT_LENGTH_UNKNOWN} is returned for other stream types or if this value is <= 0.
     * @return the {@code contentLength} value or {@link #CONTENT_LENGTH_UNKNOWN}
```